blinkm/blinkm.go

```diff
@@ -9,46 +9,52 @@ import (
 
 // Device wraps an I2C connection to a BlinkM device.
 type Device struct {
-	bus machine.I2C
+	bus     machine.I2C
+	Address uint16
```

Comment:
Why are you using `uint16` here?
I haven't yet seen an I2C device in the wild that uses 10 bit addressing. This `Device` struct is also driver specific so there is no shared interface to support here. I think it can be a `uint8`. This will also avoid having to replace `ReadRegister` with `Tx`.

(This comment also applies to all other drivers).

Reply:
The reason I replaced `ReadRegister` is actually because it creates a memory leak on AVR, since the allocated slice is never freed.

Reply:
I have looked into this a bit, but switching to `Tx` does not seem to improve the situation. I can still see a `runtime.alloc` call in the bh1750 driver after this change, for example.

After some investigation, I think it's because the built-in escape analysis of LLVM is too conservative for our purposes:
https://llvm.org/doxygen/CaptureTracking_8cpp_source.html#L332
In safe Go, there is no crazy pointer arithmetic possible like in C. Therefore, a comparison against nil can never be used to let a pointer escape. I'm not sure what the best way is to fix this.

Reply:
The reason for `Address uint16` is because the i2c interface signature is `Tx(addr uint16, w, r []byte) error`.